scripts/complete_liquidity_provision.js

```diff
@@ -131,28 +152,21 @@ module.exports = async (callback) => {
       callback("Error: Choose a smaller numBrackets, otherwise your payload will be to big for Infura nodes")
     }
 
-    let bracketAddresses
     if (argv.brackets) {
-      console.log("==> Skipping safe deployment and using brackets safeOwners")
+      console.log("==> Skipping safe deployment and using brackets")
       bracketAddresses = argv.brackets
-      await verifyBracketsWellFormed(masterSafe.address, bracketAddresses, null, null, true)
-      // Detect if provided brackets have existing orders.
-      const existingOrders = await Promise.all(
-        bracketAddresses.map(async (safeAddr) => {
-          return hasExistingOrders(safeAddr, exchange)
-        })
-      )
-      const dirtyBrackets = bracketAddresses.filter((_, i) => existingOrders[i] == true)
+      await assertBracketsWellFormedPromise
+      const dirtyBrackets = await bracketsWithExistingOrdersPromise
       if (
-        existingOrders.some((t) => t) &&
+        dirtyBrackets.length !== 0 &&
         !(await proceedAnyways(`The following brackets have existing orders:\n  ${dirtyBrackets.join()}\n`))
       ) {
         callback("Error: Existing order verification failed.")
       }
     } else {
       assert(!argv.verify, "Trading Brackets need to be provided via --brackets when verifying a transaction")
       console.log(`==> Deploying ${argv.numBrackets} trading brackets`)
-      bracketAddresses = await deployFleetOfSafes(masterSafe.address, argv.numBrackets)
+      bracketAddresses = await deployFleetOfSafes(argv.masterSafe, argv.numBrackets)
```

Comment:
~~Nice change here. Do we happen to know if there was any performance boost with regards to this change?~~

Never mind, I see what you did down below (didn't need to resolve the master safe till way later.

Reply:
Yes, this is the reason I did it. I don't actually save any time in practice by moving the resolution to the end of the file: at some point early on I need to resolve `masterOwnersPromise` anyway, which in turn requires resolving `masterSafePromise`. It might still help if we'll ever change how we resolve `masterOwnersPromise`.